File: Java/Transacao.java
public class Transacao {
    String descricao;
    double valor;
    Categoria categoria ;

    public Transacao(String descricao, double valor, Categoria categoria){
        this.descricao = descricao;
        this.valor = valor;
        this.categoria = categoria;
    }

    public void exibir(){
        System.out.println("Descrição: " + descricao + " | Valor: " + valor + " | Categoria: " + categoria.nome);
    }
}

File: Transacao.py
from dataclasses import dataclass
from categoria import Categoria

@dataclass
class Transacao:
    descricao: str
    valor: float
    categoria: Categoria

    # Metodo
    def exibir(self):
        print(f"Descrição: {self.descricao} | Valor: {self.valor} | Categoria: {self.categoria.nome}")